utils/Poller.js

```diff
@@ -11,14 +11,16 @@ class Poller {
 
     return new Promise((resolve, reject) => {
       const processHandles = [];
+      const startTime = Date.now();
 
       const timeoutHandle = setTimeout(() => {
         if (isDone) {
           return;
         }
         isDone = true;
         clearTimeouts([timeoutHandle, ...processHandles]);
-        reject(new Error(`Failed polling due to exceeding the timeout. (${name})`));
+        const totalSeconds = (Date.now() - startTime) / 1000;
```

Comment:
(minor nitpick; also might be wrong so maybe ignore me) should we convert to integer? thinking of floating issues with lots of decimals?

Reply:
`Date.now()` returns an integer value in milliseconds rather than `Date` object. The function name is confusing though 😄

Reply:
And the result has 3 decimals only because we are dividing by 1000.